bin/mark_changepoints_in_json

```diff
@@ -174,23 +174,25 @@ class Segments(object):
                 self._merge_segment(0)
 
     def get_classification(self):
-        """Return the classification for this run sequence.
-        """
-        classification = ''
-        # Flat -> no changepoints.
+        """Return the classification for this run sequence."""
+        PC = 1.001  # TODO: Factor out as a CLI option before merging with master.
+        classification = 'flat'
         if len(self.segments) == 1:
-            classification = 'flat'
-        # No steady state -> there are changes after we expect steady state.
-        elif self.segments[-1].start > (self.length - self.steady_state):
-            classification = 'no steady state'
-        # Warmup -> first segment mean is highest and last is the lowest.
-        elif (self.segments[-1].start <= (self.length - self.steady_state) and
-              self.segments[-1].mean == min(self.segments, key=lambda s: s.mean).mean):
-            classification = 'warmup'
-        # Slowdown (b) -> no warmup and no changepoints after we expect a
-        #                 steady state.
-        else:
-            classification = 'slowdown'
+            return classification
+        # > 1 segments: iterate from the penultimate to the first segment.
+        for index in xrange(len(self.segments) - 2, -1, -1):
+            if self.segments[index].mean * PC < self.segments[-1].mean:
+                if self.segments[index].end > (self.length - self.steady_state):
```

Comment:
in other words, if the last segment is too short, we have "no steady state" regardless of the mean, so we can bail out earlier.